### PR TITLE
Added basic scaffolding for NDBServer

### PR DESCRIPTION
**Pull request by manavrajvanshi**
**What this PR does / why we need it**: Adds basic scaffolding of the NDBServer CRD

**How Has This Been Tested?**:
Tested the existing workflow by manual e2e test.
image
image

**Special notes for your reviewer**: Adds basic scaffolding

**Release note**:
```release-note
Also fixed annotations in the dbcontroller
```

### CODE REVIEW

File: controllers/database_controller.go [hunk 1]
```diff
@@ -45,11 +45,13 @@ type DatabaseReconciler struct {
...
 }
 
-// The Reconcile method is where the controller logic resides.
-
 // +kubebuilder:rbac:groups=ndb.nutanix.com,resources=databases,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=ndb.nutanix.com,resources=databases/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=ndb.nutanix.com,resources=databases/finalizers,verbs=update
+// +kubebuilder:rbac:groups=core,resources=services;endpoints,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
+
```
Comment (manavrajvanshi) [inline controllers/database_controller.go:53]: Added these since these were previously hardcoded.

**Review by shenoypritika — APPROVED** [review]

**Comment by akshmish** [comment]
LGTM. One small query, I wonder why the scaffolding code did not create any test files?

**Review by akshmish — APPROVED** [review]
LGTM. Does it not create any test

**Comment by manavrajvanshi** [comment]
That is because scaffolding does not create controller specific tests, those are added manually. As discussed, adding an empty test file.

**Pull request merged**